test_rclcpp/test/node_with_name.cpp

```diff
@@ -28,8 +28,17 @@ int main(int argc, char ** argv)
   printf("Starting node with name: %s\n", node_name.c_str());
   std::cout.flush();
 
-  auto node = rclcpp::Node::make_shared(node_name);
-  rclcpp::spin(node);
+  rclcpp::Node::SharedPtr node;
+  try {
+    node = rclcpp::Node::make_shared(node_name);
+  } catch (rclcpp::exceptions::RCLError) {
+    // test may pass and send SIGINT before node finishes initializing ros2/build_cop#153
+    printf("Likely received SIGINT before node initialization finished.\n");
```

Comment:
It would be better if you printed out the error message as well, just in case it's not what you expect.

Reply:
Prints out error message in 9ea11cea045815933827a15f92b53f11a39cb271

Reply:
Just noting `stdout` is only output to the console when a test fails, so the error message is only in the `*.xunit.xml file`

```
(node_name_list) pid 22165: [&apos;/home/sloretz/workspaces/ros2-dev0/build/test_rclcpp/node_name_list&apos;, &apos;node_with_name_rmw_connext_cpp&apos;] (stderr &gt; stdout, all &gt; console)
[node_name_list] Waiting for node with name: node_with_name_rmw_connext_cpp
[node_with_name] Starting node with name: node_with_name_rmw_connext_cpp
[node_name_list]
[node_name_list] -
[node_name_list] -
[node_name_list] - node_name_list
[node_name_list] - node_with_name_rmw_connext_cpp
[node_name_list]   found expected node name
(node_name_list) rc 0
() tear down
(node_with_name) signal SIGINT
[node_with_name] RTI Data Distribution Service Non-commercial license is for academic, research, evaluation and personal use only. USE FOR COMMERCIAL PURPOSES IS PROHIBITED. See RTI_LICENSE.TXT for terms. Download free tools at rti.com/ncl. License issued to Non-Commercial User license@rti.com For non-production use only.
[node_with_name] Expires on 00-jan-00 See www.rti.com for more information.
[node_with_name] [INFO] [rclcpp]: signal_handler(signal_value=2)
[node_with_name] Ignoring RCLError: could not create client: rcl node&apos;s context is invalid, at /home/sloretz/workspaces/ros2-dev0/src/ros2/rcl/rcl/src/rcl/node.c:547
(node_with_name) rc 0
```